v2/sender.go

```diff
@@ -103,6 +104,19 @@ func SetMessageTTL(ttl time.Duration) func(msg *azservicebus.Message) error {
 	}
 }
 
+// SetTraceCarrier set the trace carrier in the message's ApplicationProperties
+// it is for tracing propagation through the ServiceBus queue messages
+func SetTraceCarrier(traceCarrier map[string]string) func(msg *azservicebus.Message) error {
```

Comment:
I'm not sure how you expect the sender to use this.

When you send a message, the trace data is in the context right?
let's split this PR in 2.
the handler looks better. I think the sender needs to be transparent. this requires too much from the user.